class.jetpack.php

```diff
@@ -4542,6 +4542,13 @@ function build_connect_url( $raw = false, $redirect = false, $from = false, $reg
 			if( is_network_admin() ) {
 				$url = add_query_arg( 'is_multisite', network_admin_url( 'admin.php?page=jetpack-settings' ), $url );
 			}
+
+			$calypso_env = self::get_calypso_env();
+
+			if ( ! empty( $calypso_env ) ) {
+				$args['calypso_env'] = $calypso_env;
```

Comment:
It doesn't look like `$args` is used in this context. Should we be doing this instead:

```suggestion
				$url = add_query_arg( 'calypso_env', $calypso_env, $url );
```